src/components/molecules/ExportProgressItem.tsx

```diff
@@ -0,0 +1,615 @@
+      setExportState((state) => {
+        return {
+          ...state,
+          ...exportOperationState,
+        }
+      })
+    },
+    []
+  )
+
+  const resetExportingStates = useCallback(() => {
+    setShowExportProgress(false)
+    updateExportState({
+      exportOperation: '',
+      exportSubOperationTitle: '',
+      exportSubOperationContent: '',
+      exportProgressValue: 0,
+    })
+    setExportErrors([])
+  }, [updateExportState])
+
+  const addExportError = useCallback((error: string) => {
+    setExportErrors((state) => [...state, error])
+  }, [])
+
+  const cancelExporting = useCallback(() => {
+    canceledRef.current = true
+    resetExportingStates()
+  }, [resetExportingStates])
+
+  const isExportingCanceled = useCallback(() => {
+    return canceledRef.current
+  }, [])
+
+  const createExportDirectories = useCallback(
+    async (folderMap: ObjectMap<FolderDoc>, rootDir: string) => {
+      for (const folderPath of keys(folderMap)) {
+        if (isExportingCanceled()) {
+          return
+        }
+
+        if (folderPath == '/' && !exportSettings.exportingStorage) {
+          continue
+        }
+
+        if (
+          !exportSettings.exportingStorage &&
+          ((exportSettings.recursive &&
+            !folderPath.startsWith(exportSettings.folderPathname)) ||
+            (!exportSettings.recursive &&
+              folderPath != exportSettings.folderPathname))
+        ) {
+          continue
+        }
+
+        const fullPath = join(
+          rootDir,
+          folderPath.substring(folderPath.indexOf(exportSettings.folderName))
+        )
+
+        updateExportState({
+          exportSubOperationTitle: `Creating directory: ${fullPath.replace(
+            rootDir,
+            '.'
+          )}`,
+        })
+        await mkdir(fullPath)
+      }
+    },
+    [
+      exportSettings.exportingStorage,
+      exportSettings.folderName,
+      exportSettings.folderPathname,
+      exportSettings.recursive,
+      isExportingCanceled,
+      updateExportState,
+    ]
+  )
+
+  const convertValueToPercentage = useCallback(
+    (value: number, maxValue: number, exportProgressMaxValue = 100) => {
+      if (maxValue === 0) {
+        return 0
+      }
+      return Math.floor((value / maxValue) * exportProgressMaxValue)
+    },
+    []
+  )
+
+  const calculateNumberOfExportingNotes = useCallback(
+    (noteMap: ObjectMap<NoteDoc>): number => {
+      const calculateNoteCountCondition = !(
+        exportSettings.recursive && noteCount !== 0
+      )
+      let exportingNoteCount = calculateNoteCountCondition ? 0 : noteCount
+      if (exportSettings.exportingStorage) {
+        exportingNoteCount = values(noteMap).length
+      } else {
+        if (calculateNoteCountCondition || exportingNoteCount === 0) {
+          for (const [, noteDoc] of entries(noteMap)) {
+            if (
+              noteDoc.trashed ||
+              (exportSettings.recursive &&
```

Comment:
Separate conditions for better readability

```ts
if (noteDoc.trashed) {
  continue
}
if (exportSettings.recursive && !noteDoc.folderPathname.startsWith) {
  continue
}

```

Reply:
Refactored!